compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -708,8 +709,18 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
         if let Some(mplace) = op.as_mplace_or_imm().left() {
             if let Some(alloc_id) = mplace.ptr().provenance.and_then(|p| p.get_alloc_id()) {
                 let mutability = match self.ecx.tcx.global_alloc(alloc_id) {
-                    GlobalAlloc::Static(_) => {
-                        self.ecx.memory.alloc_map.get(alloc_id).unwrap().1.mutability
+                    // This can only ever be the static currently being evaluated (or one of its nested allocations), as
+                    // we do not walk into other statics. See `check_safe_pointer` for details.
```

Comment:
This comment isn't true though, we can go into the nested statics of other statics.

Sounds like we want a helper function for the "is this static mutable" logic, and use that both here and up in `check_safe_pointer`?